clients/privacy-center/pages/api/gpp-js.ts

```diff
@@ -0,0 +1,52 @@
+import { promises as fsPromises } from "fs";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { CacheControl, stringify } from "cache-control-parser";
+
+import { LOCATION_HEADERS } from "~/common/geolocation";
+
+// one hour, how long the client should cache gpp-ext.js for
+const GPP_JS_MAX_AGE_SECONDS = 60 * 60;
+
+/**
+ * @swagger
+ * /gpp-ext.js:
+ *   get:
+ *     description: Returns the "gpp-ext.js" bundle for dynamic loading
+ *     responses:
+ *       200:
+ *         description: a "gpp-ext.js" script extension
+ *         content:
+ *           application/javascript:
+ *             schema:
+ *               type: string
+ *             example: |
+ *               (function(){
+ *                 // gpp.js extension bundle...
+ *               )();
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const gppJsFile = "public/lib/gpp-ext.js";
```

Comment:
FWIW: I feel like this an API route here is more complicated than we'd _want_ here, just because next.js will happily serve static files... but at the same time, it's fairly painful to force cache headers on them...

In other words, a potentially simpler way to solve this is:
1. copy that fides-gpp-ext.js file into `public/` instead of `public/lib`
2. use the default hosting of static files from next
3. add the required headers in the next.config.js or similar

Reply:
oh good point, that makes things a lot simpler!

Reply:
hm actually there seem to be some open issues about adding cache headers in the public folder: https://github.com/vercel/next.js/issues/22319

will look into some of the workarounds on monday...

Reply:
https://stackoverflow.com/questions/62077589/setting-cache-control-header-for-static-file-serving-on-nextjs-default-server